src/devices/realsense2/realsense2Tracking.cpp

```diff
@@ -0,0 +1,443 @@
+    if (!sensor->supports(option)) {
+        yError() << "realsense2Driver: The option" << rs2_option_to_string(option) << "is not supported by this sensor";
+        return false;
+    }
+
+    // To set an option to a different value, we can call set_option with a new value
+    try {
+        sensor->set_option(option, value);
+    } catch (const rs2::error& e) {
+        // Some options can only be set while the camera is streaming,
+        // and generally the hardware might fail so it is good practice to catch exceptions from set_option
+        yError() << "realsense2Driver: Failed to set option " << rs2_option_to_string(option) << ". (" << e.what() << ")";
+        return false;
+    }
+    return true;
+}
+
+static bool getOption(rs2_option option, const rs2::sensor* sensor, float& value)
+{
+    if (!sensor) {
+        return false;
+    }
+
+    // First, verify that the sensor actually supports this option
+    if (!sensor->supports(option)) {
+        yError() << "realsense2Driver: The option" << rs2_option_to_string(option) << "is not supported by this sensor";
+        return false;
+    }
+
+    // To set an option to a different value, we can call set_option with a new value
+    try {
+        value = sensor->get_option(option);
+    } catch (const rs2::error& e) {
+        // Some options can only be set while the camera is streaming,
+        // and generally the hardware might fail so it is good practice to catch exceptions from set_option
+        yError() << "realsense2Driver: Failed to get option " << rs2_option_to_string(option) << ". (" << e.what() << ")";
+        return false;
+    }
+    return true;
+}
+
+static void settingErrorMsg(const string& error, bool& ret)
+{
+    yError() << "realsense2Driver:" << error.c_str();
+    ret = false;
+}
+
+realsense2Tracking::realsense2Tracking()
+{
+}
+
+bool realsense2Tracking::pipelineStartup()
+{
+    try {
+        m_profile = m_pipeline.start(m_cfg);
+    } catch (const rs2::error& e) {
+        yError() << "realsense2Driver: failed to start the pipeline:"
+                 << "(" << e.what() << ")";
+        m_lastError = e.what();
+        return false;
+    }
+    return true;
+}
+
+bool realsense2Tracking::pipelineShutdown()
+{
+    try {
+        m_pipeline.stop();
+    } catch (const rs2::error& e) {
+        yError() << "realsense2Driver: failed to stop the pipeline:"
+                 << "(" << e.what() << ")";
+        m_lastError = e.what();
+        return false;
+    }
+    return true;
+}
+
+bool realsense2Tracking::pipelineRestart()
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    if (!pipelineShutdown())
+        return false;
+
+    return pipelineStartup();
+}
+
+bool realsense2Tracking::open(Searchable& config)
+{
+    yWarning() << "This software module is experimental.";
+    yWarning() << "It is provided with uncomplete documentation and it may be modified/renamed/removed without any notice.";
+
+    string sensor_is = "t265";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'sensor_is' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=4772294)